plugin/pkg/admission/resourcequota/admission_test.go

```diff
@@ -1439,9 +1443,331 @@ func TestAdmitLimitedResourceWithQuotaThatDoesNotCover(t *testing.T) {
 		evaluator: evaluator,
 	}
 	indexer.Add(resourceQuota)
-	newPod := validPod("not-allowed-pod", 1, getResourceRequirements(getResourceList("3", "2Gi"), getResourceList("", "")))
+	newPod := validPod("not-allowed-pod", 1, getResourceRequirements(getResourceList("3", "2Gi"), getResourceList("", "")), "")
 	err := handler.Validate(admission.NewAttributesRecord(newPod, nil, api.Kind("Pod").WithVersion("version"), newPod.Namespace, newPod.Name, api.Resource("pods").WithVersion("version"), "", admission.Create, nil))
 	if err == nil {
 		t.Fatalf("Expected an error since the quota did not cover cpu")
 	}
 }
+
+// TestAdmitLimitedScopeWithQuota verifies if a limited scope is configured the quota must cover the resource.
+func TestAdmitLimitedScopeWithCoverQuota(t *testing.T) {
```

Comment:
Could you please add a test to limit a priority class globally and allow it only in one namespace and verify that admission works correctly?

Reply:
Hi @bsalamat, following tests do this testing:
https://github.com/vikaschoudhary16/kubernetes/blob/f382229ccabd6b8b137778b674b8960a322e3718/plugin/pkg/admission/resourcequota/admission_test.go#L1705-#L1771

Reply:
The scenario that I described is not covered by those tests. The test at line 1755 is the closest one. The quota in that test has no priority class name though. It would be great if you could add a test very similar to the one at line 1705 (not 1755) and just change the pod's priority class to "another-priorityclass-name" and verify that there is no error. Thanks!

Reply:
can we add a test case that validates the same resource "pods" being limited multiple times by varying scopes?

for example:
- "pods" with matching scope "Terminating"
- "pods" with matching scope "PriorityClass In Bar"

ensure that a pod that matches neither scope is allowed as expected, but a pod that matches either is?  it looks like each test tests a single LimitedResource but having multiple input LimitedResource that varies on scope would help.

Reply:
@derekwaynecarr Added the following test cases:
```
1595:			description: "Two scopes,BestEffort and PriorityClassIN, in two LimitedResources. Neither matches pod. Pod allowed",
1624:			description: "Two scopes,BestEffort and PriorityClassIN, in two LimitedResources. Only BestEffort scope matches pod. Pod admit fails because covering quota is missing for BestEffort scope",
1653:			description: "Two scopes,BestEffort and PriorityClassIN, in two LimitedResources. Only PriorityClass scope matches pod. Pod admit fails because covering quota is missing for PriorityClass scope",
1682:			description: "Two scopes,BestEffort and PriorityClassIN, in two LimitedResources. Both the scopes matches pod. Pod admit fails because covering quota is missing for PriorityClass scope and BestEffort scope",
1711:			description: "Two scopes,BestEffort and PriorityClassIN, in two LimitedResources. Both the scopes matches pod. Quota available only for BestEffort scope. Pod admit fails because covering quota is missing for PriorityClass scope",
1753:			description: "Two scopes,BestEffort and PriorityClassIN, in two LimitedResources. Both the scopes matches pod. Quota available only for PriorityClass scope. Pod admit fails because covering quota is missing for BestEffort scope",
1795:			description: "Two scopes,BestEffort and PriorityClassIN, in two LimitedResources. Both the scopes matches pod. Quota available only for both the scopes. Pod admit success. No Error",
```

Reply:
@bsalamat Added following test case as per your suggestion:
```
1995:                         description: "From the above test case, just changing pod priority from cluster-services to another-priorityclass-name. expecting no error"
```

Reply:
Thanks! Looks good.